articles/search/search-howto-managed-identities-cosmos-db.md

```diff
@@ -22,10 +22,17 @@ You can use a system-assigned managed identity or a user-assigned managed identi
 
 * [Create a managed identity](search-howto-managed-identities-data-sources.md) for your search service.
 
-* [Assign a role](search-howto-managed-identities-data-sources.md#assign-a-role) in Azure Cosmos DB.
-
+* [Control plane role assignment](search-howto-managed-identities-data-sources.md#assign-a-role) in Azure Cosmos DB.
   For data reader access, you'll need the **Cosmos DB Account Reader** role and the identity used to make the request. This role works for all Azure Cosmos DB APIs supported by Cognitive Search. This is a control plane RBAC role. 
 
+* Data Plane Role assignment: Follow this https://learn.microsoft.com/en-us/azure/cosmos-db/how-to-setup-rbac 
+to know more about data plane role assignments for CosmosDB
+
+* *Only applicable for Cosmos Sql ApiKind*:
+  [Enforcing RBAC as the only authentication method in Azure Cosmos DB](../cosmos-db/how-to-setup-rbac.md#disable-local-auth)
+is supported for Sql Collections and follow above link to set the `disableLocalAuth` to `true` for your cosmos account.
+
+* *For Gremlin and MongoDB Collections*: 
```

Comment:
line 35, following Cosmos DB conventions for API references. Also, I wonder if we should mention preview status? It makes the exception easier to understand and also infers that we'll support in full at GA.  Here's what I suggest:

For Azure Cosmos DB for Apache Gremlin and Apache Cosmos DB for MongoDB, indexer support is currently in preview. At this time, a preview limitation exists that requires Cognitive Search to connect using keys. You can still set up a managed identity and role assignment, but Cognitive Search will only use the role assignment to get keys for the connection. This limitation means that you can't configure an RBAC-only approach if your indexers are connecting to Gremlin or MongoDB.